src/services/createRoute.ts

```diff
@@ -0,0 +1,57 @@
+import { markRaw } from 'vue'
+import { RouterView } from '@/components'
+import { CombineName } from '@/services/combineName'
+import { CombinePath } from '@/services/combinePath'
+import { CombineQuery } from '@/services/combineQuery'
+import { host } from '@/services/host'
+import { combineRoutes, CreateRouteOptions, CreateRouteOptionsWithoutParent, CreateRouteOptionsWithParent, isRouteWithParent, isRouteWithoutComponent } from '@/types/createRouteOptions'
+import { Host } from '@/types/host'
+import { Path, ToPath, toPath } from '@/types/path'
+import { Query, ToQuery, toQuery } from '@/types/query'
+import { Route } from '@/types/route'
+import { checkDuplicateKeys } from '@/utilities/checkDuplicateKeys'
+
+export function createRoute<
+  TName extends string | undefined = undefined,
+  TPath extends string | Path | undefined = undefined,
+  TQuery extends string | Query | undefined = undefined
+>(options: CreateRouteOptionsWithoutParent<TName, TPath, TQuery>): Route<TName extends string ? TName : '', Host<'', {}>, ToPath<TPath>, ToQuery<TQuery>>
+
+export function createRoute<
+  TParent extends Route,
+  TName extends string | undefined = undefined,
+  TPath extends string | Path | undefined = undefined,
+  TQuery extends string | Query | undefined = undefined
+>(options: CreateRouteOptionsWithParent<TParent, TName, TPath, TQuery>): Route<CombineName<TParent['key'], TName extends string ? TName : ''>, Host<'', {}>, CombinePath<TParent['path'], ToPath<TPath>>, CombineQuery<TParent['query'], ToQuery<TQuery>>>
+
+export function createRoute(options: CreateRouteOptions | CreateRouteOptionsWithParent<Route>): Route {
+  const routeWithComponent = addRouterViewComponentIfParentWithoutComponent(options)
```

Comment:
So we're just defaulting all routes to `router-view` components yeah?

Reply:
correct. Not 100% sure yet if that's okay or if we'll need to move that logic into the RouterView

Reply:
I think that is fine for now